pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -112,6 +113,22 @@ func IsPlacementPolicyTypeUpdated(oldPolicy, currentPolicy *placementv1beta1.Pla
 	return false
 }
 
+func IsTolerationsUpdatedOrDeleted(oldTolerations []placementv1beta1.Toleration, newTolerations []placementv1beta1.Toleration) bool {
+	for _, oldToleration := range oldTolerations {
+		found := false
+		for _, newToleration := range newTolerations {
+			if reflect.DeepEqual(oldToleration, newToleration) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			return true
+		}
+	}
+	return false
```

Comment:
would a hashmap reduce the computational complexity of this to O(n)?

Reply:
opened this PR to be able to this https://github.com/Azure/fleet/pull/685